core/emulate.c

```diff
@@ -1097,11 +1097,6 @@ em_status_t EMCALL em_decode_insn(struct em_context_t *ctxt, const uint8_t *insn
             // TODO: Should throw #UD
             return EM_ERROR;
         }
-        if (ctxt->rep == PREFIX_REPNE && !(opcode->flags & INSN_REPX)) {
```

Comment:
I'm okay with removing this check, but I'd like to see it replaced by a comment that explains why it is not done. The best argument would be "to mimic undocumented hardware behavior"--again, we need some kind of proof of that.